draft-add-requirements.md

```diff
@@ -232,23 +277,22 @@ this threat model may have limited value.
 This section lists requirements that flow from the above sections.
 
 | Requirement | Description |
-| R1.1 | Discovery MUST provide a local network the ability to announce to clients a set of, or absence of, equivalent resolvers. |
-| R1.2 | Discovery MUST provide a resolver the ability to announce to clients a set of, or absence of, equivalent resolvers. |
+| R1.1 | Discovery SHOULD provide a local network the ability to announce to clients a set of, or absence of, designated resolvers. |
+| R1.2 | Discovery SHOULD provide a resolver the ability to announce to clients a set of, or absence of, designated resolvers. |
 | R1.3 | Discovery MUST support at least one encrypted DNS protocol. |
 | R1.4 | Discovery SHOULD support all standardised encrypted DNS protocols. |
-| R2.1 | Networks MUST be able to announce one or more equivalent encrypted DNS resolvers using existing mechanisms such as DHCPv4, DHCPv6, IPv6 Router Advertisement, and the Point-to-Point Protocol. |
-| R2.2 | The format for resolver information MUST be specified such that provisioning mechanisms defined outside of the IETF can advertise encrypted DNS resolvers. |
+| R2.1 | Networks SHOULD be able to announce one or more designated encrypted DNS resolvers using existing mechanisms such as DHCPv4, DHCPv6, IPv6 Router Advertisement, and the Point-to-Point Protocol. |
+| R2.2 | The format for resolver designation SHOULD be specified such that provisioning mechanisms defined outside of the IETF can advertise encrypted DNS resolvers. |
+| R2.3 | This format SHOULD convey, at minimum, the information the client needs to make contact with each designated resolver. |
+| R2.4 | This format MAY convey more information. |
 | R3.1 | When discovery is instantiated from a resolver (R1.2), that resolver MAY be encrypted or not. |
-| R3.2 | When discovery is instantiated from a resolver (R1.2), that resolver MAY be locally or globally reachable. Both options MUST be supported. |
-| R4.1 | In a home network use case, if the local network forwarder does not offer encrypted DNS service, the ISP's encrypted DNS server information MUST be retrievable via a query sent to a local network forwarder. |
-| R4.2 | Encrypted DNS server discovery MUST NOT require any changes to DNS forwarders hosted on non-upgradable legacy network devices. |
+| R3.2 | When discovery is instantiated from a resolver (R1.2), that resolver MAY be locally or globally reachable. Both options SHOULD be supported. |
+| R4.1 | In a home network use case, if the local network resolver is a forwarder that does not offer encrypted DNS service, an upstream encrypted resolver SHOULD be retrievable via queries sent to that forwarder. |
+| R4.2 | Achieving requirement 4.1 SHOULD NOT require any changes to DNS forwarders hosted on non-upgradable legacy network devices. |
```

Comment:
I can see arguments both ways here, as a single larger requirement or two smaller requirements. Would appreciate any views from others here - I'll go with the majority.

Reply:
TBH, I'm not a fan of 4.2 anyway. I get that's a current restriction, but should the requirements for standards really take into consideration a limitation that goes away with time? CPEs can be modified, it just takes time. I don't want to require proposed standards to do this, even if some choose to do so.

If it stays, I'd like 4.1 and 4.2 to be separated so that a standard could choose to meet 4.1 without meeting 4.2.

Reply:
I prefer to keep 4.2 separate. Modern CPEs have network security services co-located and are easily upgradable (for example, see https://www.opensync.io/).

Reply:
Kept separate (for now).